### PR TITLE
Add calculated field

### PR DESCRIPTION
**Pull request by GregoireHebert**

### CODE REVIEW

File: core/serialization.md [hunk 1]
```diff
@@ -306,6 +306,71 @@ This avoids the need for extra queries to be executed when serializing the relat
...
+<?php
+
+declare(strict_types=1);
+
+namespace App\Entity;
+
+use ApiPlatform\Core\Annotation\ApiResource;
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Serializer\Annotation\Groups;
+use Symfony\Component\Validator\Constraints as Assert;
```
Comment (alanpoulain) [inline core/serialization.md:323]: Not needed.

Comment (GregoireHebert) [reply core/serialization.md:323]: I could simplify.

File: core/serialization.md [hunk 2]
```diff
@@ -306,6 +306,71 @@ This avoids the need for extra queries to be executed when serializing the relat
...
+
+/**
+ * @ApiResource(
+ *     collectionOperations={
+ *          "get" = {"normalization_context" = {"groups" = "Greeting:Collection:Get"}},
```
Comment (alanpoulain) [inline core/serialization.md:328]:
```suggestion
 *          "get"={"normalization_context"={"groups"="greeting:collection:get"}},
```

File: core/serialization.md [hunk 3]
```diff
@@ -306,6 +306,71 @@ This avoids the need for extra queries to be executed when serializing the relat
...
+    
+    private $a = 1;
+    
+    private $b = 2;
+
+    /**
+     * @var string A nice person
+     *
+     * @ORM\Column
+     * @Groups("Greeting:Collection:Get")
```
Comment (alanpoulain) [inline core/serialization.md:353]:
```suggestion
     * @Groups("greeting:collection:get")
```

File: core/serialization.md (hunk 1 above)
Comment (GregoireHebert) [reply core/serialization.md:323]: Oh no you're right I forgot to remove this one, good catch!

File: core/serialization.md [hunk 4]
```diff
@@ -306,6 +306,71 @@ This avoids the need for extra queries to be executed when serializing the relat
...
+
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return int
+     *
+     * @Groups("Greeting:Collection:Get") <- MAGIC IS HERE, You can set a group on a method.
```
Comment (alanpoulain) [inline core/serialization.md:365]:
```suggestion
     * @Groups("greeting:collection:get") <- MAGIC IS HERE, you can set a group on a method.
```

File: core/serialization.md [hunk 5]
```diff
@@ -306,6 +306,71 @@ This avoids the need for extra queries to be executed when serializing the relat
...
+     */
+    public function getSum()
+    {
+        return $this->a+$this->b;
```
Comment (alanpoulain) [inline core/serialization.md:369]:
```suggestion
        return $this->a + $this->b;
```

File: core/serialization.md [hunk 6]
```diff
@@ -306,6 +306,71 @@ This avoids the need for extra queries to be executed when serializing the relat
...
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return int
+     *
+     * @Groups("Greeting:Collection:Get") <- MAGIC IS HERE, You can set a group on a method.
+     */
+    public function getSum()
```
Comment (alanpoulain) [inline core/serialization.md:367]:
```suggestion
    public function getSum(): int
```

File: core/serialization.md [hunk 7]
```diff
@@ -306,6 +306,71 @@ This avoids the need for extra queries to be executed when serializing the relat
...
+     */
+    public $name = '';
+
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return int
```
Comment (alanpoulain) [inline core/serialization.md:363]: Not needed.

**Comment by alanpoulain** [comment]
Nice addition!

**Comment by GregoireHebert** [comment]
Done! thanks for the review @alanpoulain

**Comment by alanpoulain** [comment]
Thank you Grégoire :slightly_smiling_face:

File: core/serialization.md [hunk 8]
```diff
@@ -306,6 +306,71 @@ This avoids the need for extra queries to be executed when serializing the relat
...
+    
+    private $b = 2;
+
+    /**
+     * @var string A nice person
+     *
+     * @ORM\Column
+     * @Groups("Greeting:Collection:Get")
+     */
+    public $name = '';
```
Comment (azjezz) [inline core/serialization.md]: can we remove this since its unrelated? :)

**Pull request merged**